algebird-core/src/main/scala/com/twitter/algebird/CountMinSketch.scala

```diff
@@ -812,18 +827,44 @@ case class HeavyHitter[K](item: K, count: Long) extends java.io.Serializable
 
 object HeavyHitter {
 
-  def ordering[K: Ordering]: Ordering[HeavyHitter[K]] = Ordering.by { hh: HeavyHitter[K] => (hh.count, hh.item) }
+  import scala.reflect._
+
+  // Note that we can't define the ordering on `count` only.  We use a set to track heavy hitters, where equality means
+  // two elements are identical.  If we were to use `count` only, we would label different items in heavy hitters as
+  // identical simply because they have the same count (and such false duplicates may happen easily in practice).  For
+  // this reason we need to add another dimension to the ordering to prevent such false duplicates.  For this second
+  // dimension we use an item's default Java hash code because we do not want to introduce an `Ordering` constraint.
+  def ordering[K: ClassTag]: Ordering[HeavyHitter[K]] = {
```

Comment:
Thanks for the explanation.  Then I don't have to feel to bad about not having heard of `Hashing` before. :-)

> The benefit you note with dropWhile is even more of a win.

Just to be clear:  If we push Ordering to only TopK, then we lose `dropWhile` for TopPct (so that benefit, whether it's tiny or actually noticeable, would be gone) and we'd have to use `filter` instead.  (I made a mistake in the first comment of the Scala snippet above.)